ha/core/event_manager/event_manager.py

```diff
@@ -37,62 +49,263 @@ def get_instance():
+           list of events registered by the components. It maps the event
+           name with component name and store it in the consul. \
+           Also, maps the component with event
 
-        Args:
-            events (list): Event list.
-        """
-        if not isinstance(events, list):
-            raise InvalidEvent(f"Invalid type {events}, event type should be list")
+           Args:
+            component (str): Component Name
+            events (list): Events.
+        '''
+        # TODO: Add health monitor rules
+        Log.info(f'Received a subscribe request from {component}')
+        EventManager._validate_component(component)
+        EventManager._validate_events(events)
+        if component:
+            message_type = self._create_message_type(component)
         for event in events:
-            if event not in EVENTS:
-                raise InvalidEvent(f"Invalid event: {event}, not part of HA event list.")
-        Log.debug(f"event: {event} is valid for subscription request")
+            self._store_component_key(f'{const.COMPONENT_KEY}/{component}', event.resource_type, event.states)
+            self._store_event_key(f'{const.EVENT_KEY}', event.resource_type, event.states, comp=component)
+        Log.info('Subscription request is successfully stored into confstore')
+        return message_type
 
-    def subscribe(self, component: str, events: list) -> str:
-        """
-        Register events for the notification. It maintains list of events registered by the components.
+    def _store_component_key(self, key: str, resource_type: str, val: list) -> None:
+        '''
+           Perform actual consul store operation for component keys
+           Ex:
+           key: /cortx/ha/v1/events/subscribe/sspl
+           confstore value: ['enclosure:sensor:voltage/failed', ...]
+        '''
+        if self._confstore.key_exists(key):
+            comp_json_list = self._confstore.get(key)
 
-        Args:
-            component (str): Component Name
-            events (list): Events.
-        """
-        pass
+            # Get the compnent list from consul first
+            _, comp_list = comp_json_list.popitem()
+
+            comp_list = json.loads(comp_list)
+            for event in val:
+                new_val = resource_type + '/' + event
+                if new_val in comp_list:
+                    continue
+                # Merge the old and new component list
+                comp_list.append(new_val)
+
+            # Make sure that list is not duplicated
+            set_comp_list = list(OrderedDict.fromkeys(comp_list))
+
+            Log.debug(f'Final newly added list of subscription is: {comp_list}, \
+                       Key is: {key}')
+            # Finally update it in the json string format
+            event_string = json.dumps(set_comp_list)
+            Log.debug(f'Subscription is already done with key \
+                       as \'cortx/ha/v1/\' + f\'{key}\'. Updating the \
```

Comment:
I guess no need to mention \'cortx/ha/v1/\'

Reply:
Done